### DISCUSSION TITLE
Display prompt parameters in traces when using langfuse prompt management

### DISCUSSION THREAD

**Discussion by RichardSieg** [langfuse/langfuse#4543] [Support] [answered]
Hi there. Most of our prompts are compiled with special parameters. When looking at a trace I would like to be able to see the parameter values with which the prompt was compiled - preferably in the list of parameter tags for that trace (e.g. model, temperature etc). Thanks!

**Comment by marcklingen** [comment, marked as answer]
Thanks for sharing! Capturing prompt variables within a prompt template more naturally would be very useful, and we've discussed this a couple of times already. For a non-Langfuse prompt, it is challenging to distinguish between variables and the more fixed elements of the prompt.

If you use Langfuse Prompt Management, you can wrap the compile method with the Langfuse decorator like this and then use it throughout your code:

```python
@observe()
def compile_prompt(prompt, **kwargs):
   return prompt.compile(**kwargs)


@observe()
def main():
   prompt = langfuse.get_prompt("test")
   compiled_prompt = compile_prompt(prompt, foo="bar", i="u")
   # llm call
```

Note: This works similarly via the python/js low-level SDKs by creating a span for this.

Let me know if this helps!

**Reply by RichardSieg** [reply]
Thank you - that works! However, this somehow blows up the whole trace since I now have a trace, a span, and a generation inside the span. In my view a bit cleaner look and feel would be to have the prompt parameters inside the metadata tag list like temperature and model. Dunno whether that's applicable though. Thanks for the quick support.

**Reply by marcklingen** [reply]
Can you share your code? The prompt compile should be a sibling next to the generation within the trace

> In my view a bit cleaner look and feel would be to have the prompt parameters inside the metadata tag list like temperature and model. Dunno whether that's applicable though. Thanks for the quick support.

Agree, you can always pass the kwargs as a dict on the generation metadata. Happy to help figure this out if you can share the code of your implementation

**Reply by RichardSieg** [reply]
Ah yes good point with the metadata. I pass the prompt parameters into the generation metadata now. Thanks!

**Reply by marcklingen** [reply]
great, glad this helps!

**Reply by marcklingen** [reply]
(and thanks for moving the thread from discord to github, I think this is helpful for others as well)
